lib/src/components/NxIconDropdown/stateful/__tests__/NxStatefulIconDropdown.test.tsx

```diff
@@ -5,187 +5,233 @@
-        <div>
-          <NxStatefulIconDropdown title="stateful dropdown" />
-          <button id="test-btn">click</button>
-        </div>,
-        { attachTo: container }
-      );
-    });
-
-    act(() => {
-      // Jest/JSDom need actual events to be able to trigger effects properly
-      // See https://stackoverflow.com/questions/27557624/simulating-click-on-document-reactjs-jsdom
-      element.find('button.nx-icon-dropdown__toggle').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).toExist();
-
-    act(() => {
-      element!.find('#test-btn').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).not.toExist();
-
-    act(() => {
-      element!.find('button.nx-icon-dropdown__toggle').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).toExist();
-
-    element!.unmount();
+  it('does not open the dropdown if a click happens anywhere when the dropdown is disabled', async function() {
+    const user = userEvent.setup(),
+        { container, getByRole } = quickRender({ disabled: true }),
+        toggleBtn = getByRole('button');
+
+    expect(container.querySelector('.nx-dropdown-menu')).not.toBeInTheDocument();
+
+    await user.click(toggleBtn);
+
+    expect(container.querySelector('.nx-dropdown-menu')).not.toBeInTheDocument();
   });
 
-  it('closes the dropdown when a child is clicked - after calling the child\'s click handler', function() {
-    let element: ReactWrapper;
-
-    const childClickSpy = jest.fn();
-    act(() => {
-      element = mount(
-        <NxStatefulIconDropdown title="stateful dropdown">
-          <a id="child" onClick={childClickSpy}>Hello</a>
-        </NxStatefulIconDropdown>,
-        { attachTo: container }
-      );
-    });
-
-    act(() => {
-      element.find('button.nx-icon-dropdown__toggle').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).toExist();
-
-    act(() => {
-      element!.find('#child').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
+  it('closes the dropdown when a child is clicked - after calling the child\'s click handler', async function() {
+    const user = userEvent.setup(),
+        { container, getByRole, getByTestId } = quickRender(),
+        toggleBtn = getByRole('button');
+
+    await user.click(toggleBtn);
+
+    const menu = container.querySelector('.nx-dropdown-menu'),
+        children = getByTestId('child');
+
+    expect(menu).toBeInTheDocument();
+
+    await act(async () => { await user.click(children); });
```

Comment:
`act` is unnecessary

Reply:
When I removed the `act`, the tests also passed but there is console warning below, it seems the click fell outside the userEvent's built-in act handling, it happened on the second click which is to close the dropdown.
`Warning: An update to NxStatefulIconDropdown inside a test was not wrapped in act(...)`

Reply:
> When I removed the act, the tests also passed but there is console warning below, it seems the click fell outside the userEvent's built-in act handling, it happened on the second click which is to close the dropdown.
Warning: An update to NxStatefulIconDropdown inside a test was not wrapped in act(...)

OK that makes sense after some similar problems that I had in `NxFilterDropdown`.  Basically any time the dropdown is being closed by the global, native click handler, we sometimes still need to use `act`.